internal/cli/config/delete.go

```diff
@@ -29,31 +31,38 @@ type DeleteOpts struct {
 	*cli.DeleteOpts
 }
 
-func (opts *DeleteOpts) Run() error {
-	if !opts.Confirm {
-		return nil
+func (opts *DeleteOpts) executeLogout() error {
+	// Get the current executable path
+	executable, err := os.Executable()
+	if err != nil {
+		return fmt.Errorf("failed to get executable path: %w", err)
 	}
 
-	if err := config.SetName(opts.Entry); err != nil {
-		return err
-	}
+	// Execute: atlas auth logout --profile <profile-name> --force
+	cmd := exec.Command(executable, "auth", "logout", "--profile", opts.Entry, "--force")
+	cmd.Stdout = os.Stdout
+	cmd.Stderr = os.Stderr
 
-	if err := config.Delete(); err != nil {
-		return err
+	return cmd.Run()
+}
+
+func (opts *DeleteOpts) Run() error {
+	if !opts.Confirm {
+		return nil
 	}
-	fmt.Printf(opts.SuccessMessage(), opts.Entry)
-	return nil
+	return opts.executeLogout()
 }
 
 func DeleteBuilder() *cobra.Command {
 	opts := &DeleteOpts{
 		DeleteOpts: cli.NewDeleteOpts("Profile '%s' deleted\n", "Profile not deleted"),
 	}
 	cmd := &cobra.Command{
-		Use:     "delete <name>",
-		Aliases: []string{"rm"},
-		Short:   "Delete a profile.",
-		Args:    require.ExactArgs(1),
+		Use:        "delete <name>",
+		Aliases:    []string{"rm"},
+		Short:      "Delete a profile.",
+		Args:       require.ExactArgs(1),
+		Deprecated: "Please use the 'atlas auth logout' command instead.",
```

Comment:
Q: if the page is deleted, will users actually see this deprecation message? perhaps it is displayed in help output? Usually when we deprecate features we keep the page around for a while but add a banner so users get advance guidance on how they need to adjust their workflows accordingly.

My recommendation would be to keep this page and we can add a banner on our end to give users a heads-up. What are your thoughts?

Reply:
Hey @jwilliams-mongo , yes the deprecation warning would show up on call and on help.

Unfortunately, the removal is inherited from the cobra framework, I'll check if they have any way to override it but if they don't it might not be worth the effort to keep the page.